### PR TITLE
Have working ERC721s with tests

### PR DESCRIPTION
**Pull request by excalq**
Tests and deploy scripts are WIP

### CODE REVIEW

File: evm/.env.example [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Networks
+# Partavate Geth Account
+ETH_PUBLIC_KEY=0x51F5c4154854c7Dc64642a8eE79DAAdB3da57569
```
Comment (Zaskoda) [inline evm/.env.example:3]: Is this used somewhere? I don't see where.

File: evm/.env.example [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+RINKEBY_GETH_URI=http://rinkeby.partavate.com/rpc
+RINKEBY_GETH_PRIVATE_KEY=
```
Comment (Zaskoda) [inline evm/.env.example:6]: Same here, we're not deploying anything to Rinkeby, are we?

File: evm/contracts/MultiverseShip.L1.sol [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+// Moonbase Alpha L1 Contract for O8 MultiverseShip
+// Note, this conforms to IL2StandardERC721, though we're inheriting it, due to event defs present in MultiverseShip
+contract MultiverseShip_L1 is MultiverseShip {
+
+    // For Boba's IL2StandardERC721 compatibility (ideally this would be a mapping value at [chainId])
+    address public l2Contract;
+
+    // For Boba L2 Bridge Compatibility
+    function supportsInterface(bytes4 _interfaceId) public view override(MultiverseShip) returns (bool) {
```
Comment (Zaskoda) [inline evm/contracts/MultiverseShip.L1.sol:22]: I don't see the differences between these methods in the L1 and L2 contracts, so why not have them in the parent contract?

File: evm/hardhat.config.ts [hunk 4]
```diff
@@ -1,8 +1,70 @@
...
+import "@nomiclabs/hardhat-etherscan";
 import { HardhatUserConfig } from "hardhat/config";
 import "@nomicfoundation/hardhat-toolbox";
+import dotenv from "dotenv";
+
+dotenv.config();
 
 const config: HardhatUserConfig = {
   solidity: "0.8.9",
+  networks: {
```
Comment (Zaskoda) [inline evm/hardhat.config.ts:11]: There's another file in the project with network information like this and I wonder if we want to somehow try to combine them. Perhaps not for this project, but something to consider in enterprise.

File: evm/scripts/deploy.ts [hunk 5]
```diff
@@ -1,18 +1,12 @@
...
-  const lockedAmount = ethers.utils.parseEther("1");
+  await ship.deployed();
 
-  const Lock = await ethers.getContractFactory("Lock");
-  const lock = await Lock.deploy(unlockTime, { value: lockedAmount });
-
-  await lock.deployed();
-
-  console.log("Lock with 1 ETH deployed to:", lock.address);
+  console.log("Ship deployed to:", ship.address);
```
Comment (Zaskoda) [inline evm/scripts/deploy.ts:9]: Looks like we still need to have a task for recording the contract address once deployed?

**Review by Zaskoda — APPROVED** [review]
lgtm

File: evm/contracts/MultiverseShip.L1.sol (hunk 3 above)
Comment (excalq) [reply evm/contracts/MultiverseShip.L1.sol:22]: They are subtly different. Notice the use of `IL1StandardERC721` vs `IL2StandardERC721` (1 vs 2) and `IL1StandardERC721.l2contract` and `IL2StandardERC721.l1contract`?

File: evm/hardhat.config.ts (hunk 4 above)
Comment (excalq) [reply evm/hardhat.config.ts:11]: Maybe we should build our own npm/yarn packages with network/RPC information for our projects, and just include them in each project.

File: evm/scripts/deploy.ts (hunk 5 above)
Comment (excalq) [reply evm/scripts/deploy.ts:9]: Yes, I'll be adding that to the next PR.

**Pull request merged**